### PR TITLE
Implement it

### PR DESCRIPTION
**Pull request by nicolaslt**
Should be it working, give or take.
Typescript is happy with the signature, just not plugged it to a full project just yet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,4 +4,46 @@
...
+
+### At a glance
+#### Javascript
+```js
+const myMetric = require('prom-client').Histogram()
+const startTimer = require('@fresh8/prom-client-timer-unit')
+
+const stopTimer = startTimer(myMetric, 1e-3)
+// [...] your code doing things
+stopTimer() //will calculate the time elapsed and register it in `myMetric`
```
Comment (TobyColeman) [inline README.md]: space after `//`

File: README.md [hunk 2]
```diff
@@ -4,4 +4,46 @@
...
+
+#### Typescript
+```ts
+import * as promClient from 'prom-client'
+import * as startTimer from '@fresh8/prom-client-timer-unit'
+
+const myMetric = promClient.Histogram()
+const stopTimer = startTimer(myMetric, 1e-3)
+// [...] your code doing things
+stopTimer() //will calculate the time elapsed and register it in `myMetric`
```
Comment (TobyColeman) [inline README.md]: space after `//`

File: README.md [hunk 3]
```diff
@@ -4,4 +4,46 @@
...
+```
+
+### API
+#### startTimer(metric, factor, labels)
+* **metric** : your metric straight from `prom-client` (e.g: `new promClient.Histogram(opts)`)
```
Comment (TobyColeman) [inline README.md]: I think the might read better as 
> The prom-client metric, e.g. `new promClient.Histogram(opts)`

File: README.md [hunk 4]
```diff
@@ -4,4 +4,46 @@
...
+* **factor** (optional): The factor to be applied to the observed number of seconds. Examples: `1e-3` for milliseconds, `60` for minutes. Defaults to `1`
+* **labels** (optional): labels to apply to the observation. This is to follow the same signature as the original `startTimer` function in `prom-client`. Defaults to `{}`
```
Comment (TobyColeman) [inline README.md]: Labels

File: README.md [hunk 5]
```diff
@@ -4,4 +4,46 @@
...
+* returns: stopTimer
+
+Start counting time using `process.hrtime`.
+
+#### stopTimer(labels)
+* **labels** : labels to apply to the observation. They will be merged with the labels given to `startTimer` (if any). Defaults to `{}`
```
Comment (TobyColeman) [inline README.md]: Labels

File: README.md [hunk 6]
```diff
@@ -4,4 +4,46 @@
...
+
+Stops the timer and record the elapsed time in the metric given to `startTimer`, dividing the default time (in seconds) by the provided `factor`.
```
Comment (TobyColeman) [inline README.md]: > Stops the timer, setting the elapsed time to `seconds/factor`?

Comment (nicolaslt) [reply README.md]: Commited alternative proposition.

**Pull request merged**